pkg/wrangler/context.go

```diff
@@ -122,6 +123,11 @@ type Context struct {
 	PeerManager         peermanager.PeerManager
 	Provisioning        provisioningv1.Interface
 
+	MGMTWithAgent    func(string) managementv3.Interface
```

Comment:
So the challenge here was the factories these interfaces come from were only accessible in NewContext. I tried multiple approaches and landed on what I think is the best one. Basically I am putting the factories on the struct as non exported fields then using them to call their own WithAgent. I do this in Context's own WithAgent function that I added to conform more to what is happening for the norman portion.

Wrangler was leveraging a struct and function from steve that handles a subsection of the controllers. This made it difficult as using them made the factories that the controller came from inaccesible too. I could just make new factories in the WithAgent function but I felt the use of the steve logic was unnecessary. It may have saved time originally but it's different than how all the other controllers are configured so I removed the steve logic and made the factories where all the others were created and added them to the context struct as well.

Reply:
Between having a function as a member variable and storing the factories as member variables. I rather just store the factories since. Ultimately they are being stored in the struct for both approaches. The latter is just a little clearer though more verbose.